sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala

```diff
@@ -755,6 +755,31 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     val genCode = StaticInvoke(TestFun.getClass, IntegerType, "foo", arguments).genCode(ctx)
     assert(!genCode.code.toString.contains("boxedResult"))
   }
+
+  test("StaticInvoke call return `any` method") {
+    val cls = TestStaticInvokeReturnAny.getClass
+    Seq((0, IntegerType, true), (1, IntegerType, true), (2, IntegerType, false)).foreach {
```

Comment:
- case 0 - `(0, IntegerType, true)` will hit line `325` of `objects.scala`
- case 1 - `(1, IntegerType, true)` will hit line `332` of `objects.scala`
- case 2 - `(2, IntegerType, true)` will hit line `340` of `objects.scala`

Reply:
This UT can hit three points where `cast` is added.